### ISSUE TITLE
twisted.runner.procmon not even importable on windows

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2389]
|<img alt="seletz's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| seletz reported|
|-|-|
|Trac ID|trac#2389|
|Type|defect|
|Created|2007-01-11 15:00:03Z|

The module twisted.runner.procmon isn't importable on windows because it imports
twisted.internet.process directly. twisted.internet.process is not used further,
just an exception class is used from it, which isn't even defined there.

A simple patch to make the twisted.runner.procmon importable is attached.

Attachments:

* [procmon-win-importable.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/9e4/9e434fbc18017b851fbe8eda491f202f583e9d7a/c85777c815aa5a2512ea165989d41f6989b9e201.patch) (1381 bytes) - added by seletz on 2007-01-11 15:03:34Z - a patch to make the module importable by removing the t.i.process dependency.
* [procmon-win-importable-take-2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/9e4/9e434fbc18017b851fbe8eda491f202f583e9d7a/02f43fda812c1192359977b08879a9419a7b0bfe.patch) (1549 bytes) - added by seletz on 2007-01-11 15:44:58Z - 2nd version. SIGKILL/SIGTERM import hack.

<details><summary>Searchable metadata</summary>

```
trac-id__2389 2389
type__defect defect
reporter__seletz seletz
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1168527603000000 1168527603000000
changetime__1240063342000000 1240063342000000
version__None None
owner__seletz seletz
cc__teratorn
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="seletz's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|seletz commented|
|-|-|

btw, there's the submit button missing on the user configuration page in trac where I would enter my email address.

stefan dot eletzhofer at googlemail dot com

x_X

**Comment by twisted-trac** [comment]
|<img alt="seletz's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|seletz commented|
|-|-|

With the "fixes" in the last patch applied the procmon main "test" works for me
using the cygwin bash instead on /bin/sh.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to seletz|
|-|-|

Could you add unit tests for twisted.runner.procmon as well?  At least for the specific bugs which these patches address?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

This will be fixed as part of [#3691](https://github.com/twisted/twisted/issues/3691).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set status to `closed`|
|-|-|
